tests/slsa_analyzer/build_tool/test_gradle.py

```diff
@@ -86,3 +87,32 @@ def test_get_group_ids_separate_projects(tmp_path: Path, gradle_tool: Gradle) ->
         "io.micronaut.foo",
         "io.micronaut.bar",
     }
+
+
+@pytest.mark.parametrize(
+    ("timeout", "expected"),
+    [("0", set()), ("invalid", {"io.micronaut"})],
```

Comment:
Nit: I personally prefer using trailing comma here to have each test case on a separate line.
```suggestion
    [
        ("0", set()),
        ("invalid", {"io.micronaut"}),
    ],
```

Reply:
Interesting. I was expecting `black` to pick that up. I guess it's not supporting the pytest annotations?

Reply:
I think `black` doesn't usually add trailing commas automatically by itself, except for certain cases (e.g. line is too long).

Reply:
See commit [fd880ec](https://github.com/oracle/macaron/pull/446/commits/fd880ec99dc041209783be5fc8980d7f0163966c)

Reply:
Uh, I don't think it is properly formatted now. 😅 The whole decorator collapsed onto one line.

Reply:
Blame `black`!